### ISSUE TITLE
[Bug] Default configuration is used when config file specified via -config not found

### ISSUE THREAD

**Issue by beatcracker** [GitTools/GitVersion#2848]
**Describe the bug**

When invalid file name passed to the `-config` switch (missing file, etc...) no error message is shown and GitVersion silently assumes default configuration.

## Expected Behavior

If `-config` switch is used and specified file not found an error message should be shown.

## Actual Behavior

GitVersion silently assumes default configuration. This breaks the principle of least astonishment and makes it hard to debug issues with configuration, especially in CI.

## Steps to Reproduce

The GitVersion will generate version using default configuration if the specified config file doesn't exist.

```shell
mkdir ./repro
cd ./repro
git init
dotnet tool install --global GitVersion.Tool
dotnet-gitversion -config foobar.xyz
```

## Your Environment

- Operating System and version: `Windows 10`
- GitVesion: `5.7.0+Branch.main.Sha.8d177c6d666c8eeb1c6a6a2c71fd4b78741137d2`

---

I think similar issue related to config file handling was raised already, but unfortunely I can't find it, since `config` + everything that I was able to come up with is too generic terms for GitHub issue search.

**Comment by asbjornu** [comment]
I agree GitVersion should fail, but this can be considered a breaking change and should probably be postponed to v6?

**Comment by beatcracker** [comment]
Well, there are probably people out there who rely on this behavior ([Hyrum's Law](https://www.hyrumslaw.com/)), but I'm not sure if keeping it worth the trouble.

For instance I'm currently trying to debug why GitVersion refuses to read my config file specified via `-config` switch and I had  to put a `GitVersion.yaml` with garbage inside it into the repo root so GitVersion would actually fail and I could see that my custom config is not being used. And this happens locally. In CI that would be much bigger PITA.

So I can think of several improvements/fixes for this:

* Breaking change: fail if specified config not found
* Minor change: introduce additional flag that would enable `fail` behavior.
* QoL improvement: log full config name to the console output

**Comment by HHobeck** [comment]
> When invalid file name passed to the -config switch (missing file, etc...) no error message is shown and GitVersion silently assumes default configuration.

I think this statement is confusing a little bit and is not correct: GitVersion always takes the default GitVersion configuration in code. As a user you are able to override this configuration via parameter or defining the configuration file but the base is still the default configuration in our case the gitflow workflow.

**Comment by HHobeck** [comment]
> Well, there are probably people out there who rely on this behavior ([Hyrum's Law](https://www.hyrumslaw.com/)), but I'm not sure if keeping it worth the trouble.

In my opinion it makes no sense to ensure an undocumented feature/behaviour (at least for the next major version) because it was there in the past. Defining a configuration file which is not existing should result in a error for sure.

**Comment by asbjornu** [comment]
Sure, but we are also trying to close down the development of v5 to focus on v6. So unless it's very important, such as a security fix or similar, we would prefer to target everything for v6 from now on.
